executor/runtime/docker/docker_ssh.go

```diff
@@ -0,0 +1,162 @@
+package docker
+
+import (
+	"archive/tar"
+	"fmt"
+	"io/ioutil"
+
+	"strings"
+
+	"github.com/Netflix/titus-executor/config"
+	runtimeTypes "github.com/Netflix/titus-executor/executor/runtime/types"
+	"github.com/aws/aws-sdk-go/aws/arn"
+)
+
+const sshdConfig = `
+# See the sshd_config(5) manpage for details
+
+# What ports, IPs and protocols we listen for
+Port 7522
+# Use these options to restrict which interfaces/protocols sshd will bind to
+#ListenAddress ::
+#ListenAddress 0.0.0.0
+Protocol 2
+# HostKeys for protocol version 2
+HostKey /titus/sshd/etc/ssh/ssh_host_rsa_key
+HostKey /titus/sshd/etc/ssh/ssh_host_ecdsa_key
+HostKey /titus/sshd/etc/ssh/ssh_host_ed25519_key
+
+Ciphers chacha20-poly1305@openssh.com,aes256-gcm@openssh.com,aes128-gcm@openssh.com,aes256-ctr,aes192-ctr,aes128-ctr
+MACs hmac-sha2-512-etm@openssh.com,hmac-sha2-256-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-512,hmac-sha2-256,umac-128@openssh.com
+
+# Logging
+SyslogFacility AUTH
+LogLevel INFO
+
+# Authentication:
+LoginGraceTime 120
+PermitRootLogin without-password
+StrictModes yes
+
+PubkeyAuthentication yes
+#AuthorizedKeysFile	%h/.ssh/authorized_keys
+
+TrustedUserCAKeys /titus/etc/ssh/trusted_user_ca_keys.pub
+AuthorizedPrincipalsFile /titus/etc/ssh/authorized_principals_%u
+
+# Don't read the user's ~/.rhosts and ~/.shosts files
+IgnoreRhosts yes
+# similar for protocol version 2
+HostbasedAuthentication no
+# Uncomment if you don't trust ~/.ssh/known_hosts for RhostsRSAAuthentication
+#IgnoreUserKnownHosts yes
+
+# To enable empty passwords, change to yes (NOT RECOMMENDED)
+PermitEmptyPasswords no
+
+# Change to yes to enable challenge-response passwords (beware issues with
+# some PAM modules and threads)
+ChallengeResponseAuthentication no
+
+# Change to no to disable tunnelled clear text passwords
+PasswordAuthentication no
+
+# Kerberos options
+#KerberosAuthentication no
+#KerberosGetAFSToken no
+#KerberosOrLocalPasswd yes
+#KerberosTicketCleanup yes
+
+# GSSAPI options
+#GSSAPIAuthentication no
+#GSSAPICleanupCredentials yes
+
+X11Forwarding no
+X11DisplayOffset 10
+PrintMotd no
+PrintLastLog yes
+TCPKeepAlive yes
+#UseLogin no
+
+#MaxStartups 10:30:60
+#Banner /etc/issue.net
+
+# Allow client to pass locale environment variables
+AcceptEnv LANG LC_*
+
+Subsystem sftp /titus/sshd/usr/lib64/misc/sftp-server
+
+PidFile /run/sshd.pid
+`
+
+func addContainerSSHDConfig(c *runtimeTypes.Container, tw *tar.Writer, cfg config.Config) error {
+	iamProfileARN, err := c.GetIamProfile()
+	if err != nil {
+		return err
+	}
+	iamProfile, err := arn.Parse(iamProfileARN)
+	if err != nil {
+		return err
+	}
+
+	caData, err := ioutil.ReadFile(c.Config.ContainerSSHDCAFile)
```

Comment:
Is this shared between all of the containers?

Reply:
So, the configuration can be set at executor start up time to point to a different file. It gets copied into the container at startup time, so if the file changes, the container wont reflect it.

Reply:
I guess, different question, what is "this" that you refer to?

Reply:
The CA file that this is pointing to.  I'm wondering if it's one CA file used for all ssh containers on the host, or if it's intended to change per SSH container.

Reply:
It's configurable, but by default it's one CA file for all containers on the host (we can override it via config, or mutable deploy)